apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -729,7 +745,12 @@ normalize_search_text(Text, WordSeparator) ->
 packet_to_search_body(Module, Host, Packet) ->
     case has_full_text_search(Module, Host) of
         true ->
-            BodyValue = xml:get_tag_cdata(xml:get_subtag(Packet, <<"body">>)),
+            BodyValue = case exml_query:subelement(Packet, <<"body">>) of
```

Comment:
This probably could be simplified with `exml_query:path(Packet, [{element, <<"body">>}, cdata], <<"">>)`